src/scimlfunctions.jl

```diff
@@ -4067,8 +4067,8 @@ function BVPFunction{iip, specialize, twopoint}(f, bc;
     end
 
     if twopoint
-        if iip && (bcresid_prototype === nothing || length(bcresid_prototype) != 2)
-            error("bcresid_prototype must be a tuple / indexable collection of length 2 for a inplace TwoPointBVPFunction")
+        if iip && bcresid_prototype !== nothing
```

Comment:
how is this different?

Reply:
It will first test if ```bcresid_prototype``` is ```nothing```, then check the length of ```bcresid_prototype```. The original one ```bcresid_prototype === nothing || length(bcresid_prototype) != 2``` will cause error when ```bcresid_prototype``` is ```nothing```

Reply:
no it cannot, otherwise how would you know how many conditions are on the left vs right?

Reply:
Oh, I see, thanks